staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types.go

```diff
@@ -87,6 +87,14 @@ type CustomResourceDefinitionSpec struct {
 	// +listType=atomic
 	AdditionalPrinterColumns []CustomResourceColumnDefinition `json:"additionalPrinterColumns,omitempty" protobuf:"bytes,8,rep,name=additionalPrinterColumns"`
 
+	// selectableFields specifies paths to fields that may be used as field selectors.
+	// See https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors
+	//
+	// +featureGate=CustomResourceFieldSelectors
+	// +optional
+	// +listType=atomic
+	SelectableFields []SelectableField `json:"selectableFields,omitempty" protobuf:"bytes,11,rep,name=selectableFields"`
```

Comment:
refresh my memory.  This happened because we support, "if all versions are the same schema, don't nest in version" and even though this is a new field we still serialize into v1beta1 for storage, so this becomes necessary?

Reply:
Yep. v1beta1 has the top level fields that apply to all versions if the data is the same for all versions. A lot of the complexity of this PR is to support this. BTH, I'm mirrored the behavior of `AdditionalPrinterColumns` when adding `SelectableFields` to avoid trying to rethink it all.